### PR TITLE
Amplify Android API Relational Models

### PR DESCRIPTION
**Pull request by tylerjroach**
#### Description of changes:

Added new page under GraphQL API for Android showing support for Relational models with Lazy Loading and Custom Selection Set usage

### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [x] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [x] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [x] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by renebrandel — APPROVED** [review]

**Review by atierian — APPROVED** [review]
Looks good. We should add a section outlining steps to take for customers with existing schemas. That can be a follow up / may be better suited as a guide or blog.

File: src/pages/lib/graphqlapi/relational/q/platform/[platform].mdx [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+
+```kotlin
+ModelQuery.get<Comment, CommentPath>(
+    Comment::class.java, 
+    "c1"
+) { commentPath ->
+    includes(commentPath.post.comments)
+}
+```
+The network request for post includes the comments, eagerly loading the first page of comments in a single network call.
```
Comment (ankpshah) [inline src/pages/lib/graphqlapi/relational/q/platform/[platform].mdx:807]: Same as above comment, should it be - 
The network request for comments includes the post, eagerly loading the first page of comments in a single network call?

File: src/pages/lib/graphqlapi/relational/q/platform/[platform].mdx [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+ModelQuery.get<Comment, CommentPath>(Comment::class.java, "c1") { commentPath ->
+    includes(commentPath.post.comments)
+}
+```
+The network request for post includes the comments, eagerly loading the first page of comments in a single network call.
```
Comment (ankpshah) [inline src/pages/lib/graphqlapi/relational/q/platform/[platform].mdx:795]: Should this be - 
The network request for comments includes the post, eagerly loading the first page of comments in a single network call?

Comment (tylerjroach) [reply src/pages/lib/graphqlapi/relational/q/platform/[platform].mdx]: Good catch. Corrected.

**Review by ankpshah — APPROVED** [review]

**Review by atierian — APPROVED** [review]

**Review by renebrandel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**